### PR TITLE
r/aws_opsworks_custom_layer - add cloudwatch configuration + disappears test

### LINKED ISSUE #2656: Opsworks custom layer cloudwatch logs configuration
Hi,

I can see the option to configure the Cloudwatch Logs options in a layer is missing in Terraform.
It will be useful to add it.

### LINKED ISSUE #15010: CloudWatch Logs integration  Missing on OpsWorks Layers
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When you create a Layer on Opsworks, e.g. a Custom Layer <https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opsworks_custom_layer>, it's missing the option to activate Cloudwatch Logs.

The option is the following:

image

### Affected Resource

* aws_opsworks_*_layer

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #2656
Closes #15010
Relates #13826

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_opsworks_custom_layer - add cloudwatch configuration
resource_aws_opsworks_custom_layer - add plan time validation to `ebs_volume.type`
```

Output from acceptance testing:

```
$ make testacc TESTS=TestAccOpsWorksCustomLayer_ PKG=opsworks
--- PASS: TestAccOpsWorksCustomLayer_basic (112.16s)
--- PASS: TestAccOpsWorksCustomLayer_disappears (125.37s)
--- PASS: TestAccOpsWorksCustomLayer_noVPC (221.92s)
--- PASS: TestAccOpsWorksCustomLayer_cloudwatch (280.16s)
--- PASS: TestAccOpsWorksCustomLayer_tags (364.21s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Validated acceptance test:

```console
$ AWS_DEFAULT_REGION=us-east-1 make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSOpsworksCustomLayer_cloudwatch'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSOpsworksCustomLayer_cloudwatch -timeout 120m
=== RUN   TestAccAWSOpsworksCustomLayer_cloudwatch
=== PAUSE TestAccAWSOpsworksCustomLayer_cloudwatch
=== CONT  TestAccAWSOpsworksCustomLayer_cloudwatch
--- PASS: TestAccAWSOpsworksCustomLayer_cloudwatch (59.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	60.014s
```

**Comment by DrFaust92** [comment]
rebased:

```
--- PASS: TestAccAWSOpsworksCustomLayer_disappears (75.28s)
--- PASS: TestAccAWSOpsworksCustomLayer_basic (83.32s)
--- PASS: TestAccAWSOpsworksCustomLayer_noVPC (116.26s)
--- PASS: TestAccAWSOpsworksCustomLayer_cloudwatch (151.43s)
--- PASS: TestAccAWSOpsworksCustomLayer_tags (203.47s)
```

File: aws/opsworks_layers.go [hunk 1]
```diff
@@ -214,6 +222,101 @@ func (lt *opsworksLayerType) SchemaResource() *schema.Resource {
...
+								},
+								"log_group_name": {
+									Type:     schema.TypeString,
+									Required: true,
+								},
+								"multiline_start_pattern": {
+									Type:     schema.TypeString,
+									Optional: true,
+								},
+								"timezone": {
```
Comment (ewbankkit) [inline aws/opsworks_layers.go:309]: Strictly this should be `time_zone` as the API field is name `TimeZone`, as for `aws_mq_broker`, not `Timezone` as it is for `aws_db_instance`.

File: aws/opsworks_layers.go [hunk 2]
```diff
@@ -214,6 +222,101 @@ func (lt *opsworksLayerType) SchemaResource() *schema.Resource {
...
+								},
+								"initial_position": {
+									Type:         schema.TypeString,
+									Optional:     true,
+									Default:      opsworks.CloudWatchLogsInitialPositionStartOfFile,
+									ValidateFunc: validation.StringInSlice(opsworks.CloudWatchLogsInitialPosition_Values(), false),
+								},
+								"log_group_name": {
+									Type:     schema.TypeString,
+									Required: true,
```
Comment (ewbankkit) [inline aws/opsworks_layers.go:303]: Can validate with `validateLogGroupName`.

File: aws/opsworks_layers.go (hunk 1 above)
Comment (ewbankkit) [inline aws/opsworks_layers.go:309]: Strictly this should be `time_zone` as the API field is name `TimeZone`, as for `aws_mq_broker`, not `Timezone` as it is for `aws_db_instance`.

```suggestion
								"time_zone": {
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSOpsworksCustomLayer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSOpsworksCustomLayer_ -timeout 180m
=== RUN   TestAccAWSOpsworksCustomLayer_basic
=== PAUSE TestAccAWSOpsworksCustomLayer_basic
=== RUN   TestAccAWSOpsworksCustomLayer_tags
=== PAUSE TestAccAWSOpsworksCustomLayer_tags
=== RUN   TestAccAWSOpsworksCustomLayer_noVPC
=== PAUSE TestAccAWSOpsworksCustomLayer_noVPC
=== RUN   TestAccAWSOpsworksCustomLayer_cloudwatch
=== PAUSE TestAccAWSOpsworksCustomLayer_cloudwatch
=== RUN   TestAccAWSOpsworksCustomLayer_disappears
=== PAUSE TestAccAWSOpsworksCustomLayer_disappears
=== CONT  TestAccAWSOpsworksCustomLayer_basic
=== CONT  TestAccAWSOpsworksCustomLayer_cloudwatch
=== CONT  TestAccAWSOpsworksCustomLayer_noVPC
=== CONT  TestAccAWSOpsworksCustomLayer_tags
=== CONT  TestAccAWSOpsworksCustomLayer_disappears
--- PASS: TestAccAWSOpsworksCustomLayer_disappears (43.27s)
--- PASS: TestAccAWSOpsworksCustomLayer_basic (49.63s)
--- PASS: TestAccAWSOpsworksCustomLayer_noVPC (60.84s)
--- PASS: TestAccAWSOpsworksCustomLayer_cloudwatch (74.55s)
--- PASS: TestAccAWSOpsworksCustomLayer_tags (99.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	100.103s
```

File: aws/opsworks_layers.go (hunk 2 above)
Comment (DrFaust92) [reply aws/opsworks_layers.go:304]: done

File: aws/opsworks_layers.go (hunk 1 above)
Comment (DrFaust92) [reply aws/opsworks_layers.go]: done

**Comment by DrFaust92** [comment]
rebased:

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccOpsWorksCustomLayer_ PKG=opsworks
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/opsworks/... -v -count 1 -parallel 20 -run='TestAccOpsWorksCustomLayer_'  -timeout 180m
=== RUN   TestAccOpsWorksCustomLayer_basic
=== PAUSE TestAccOpsWorksCustomLayer_basic
=== RUN   TestAccOpsWorksCustomLayer_tags
=== PAUSE TestAccOpsWorksCustomLayer_tags
=== RUN   TestAccOpsWorksCustomLayer_noVPC
=== PAUSE TestAccOpsWorksCustomLayer_noVPC
=== RUN   TestAccOpsWorksCustomLayer_cloudwatch
=== PAUSE TestAccOpsWorksCustomLayer_cloudwatch
=== RUN   TestAccOpsWorksCustomLayer_disappears
=== PAUSE TestAccOpsWorksCustomLayer_disappears
=== CONT  TestAccOpsWorksCustomLayer_basic
=== CONT  TestAccOpsWorksCustomLayer_cloudwatch
=== CONT  TestAccOpsWorksCustomLayer_noVPC
=== CONT  TestAccOpsWorksCustomLayer_disappears
=== CONT  TestAccOpsWorksCustomLayer_tags
--- PASS: TestAccOpsWorksCustomLayer_disappears (52.89s)
--- PASS: TestAccOpsWorksCustomLayer_basic (70.77s)
--- PASS: TestAccOpsWorksCustomLayer_noVPC (80.00s)
--- PASS: TestAccOpsWorksCustomLayer_cloudwatch (108.23s)
--- PASS: TestAccOpsWorksCustomLayer_tags (116.01s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/opsworks	122.059s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**